Add Portainer docs (#39268)
home-assistant/home-assistant.github.io: source/_integrations/portainer.markdown

---
title: Portainer
description: Instructions on how to integrate Portainer with Home Assistant.
ha_category:
  - Binary sensor
ha_release: 0.41
ha_iot_class: Cloud Polling
ha_codeowners:
  - '@erwindouna'
ha_domain: portainer
ha_config_flow: true
ha_platforms:
  - binary_sensor
ha_integration_type: integration
---

> Portainer is a lightweight management UI which allows you to easily manage your Docker containers, images, networks and volumes. It works on every Docker host or Swarm cluster.

Reviewer: ```suggestion
The Portainer {% term integration %} is used as an interface to the [Portainer API](https://docs.portainer.io/api).
Portainer is a lightweight management UI that allows you to easily manage your Docker containers, images, networks, and volumes. It works on every Docker host or Swarm cluster.
```

> The Portainer integration platform is used as an interface to the [Portainer API](https://docs.portainer.io/api).

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
The Portainer API provides a way to manage Docker containers, images, networks, and volumes. It allows you to interact programmatically with your Docker host or Swarm cluster.

```

> {% note %}

Reviewer: ```suggestion
## Supported functionality

{% note %}
```

This integration is currently being actively developed and tested. More platforms and features will be added in.
{% endnote %}

There is currently support for the following device types within Home Assistant:

- Binary sensor - for monitoring the status of Portainer services.

> The Portainer API provides a way to manage Docker containers, images, networks, and volumes. It allows you to interact with your Docker host or Swarm cluster programmatically.

Reviewer: ```suggestion
```

---
title: Portainer
description: Instructions on how to integrate Portainer with Home Assistant.
ha_category:
  - Binary sensor
ha_release: 0.41
ha_iot_class: Cloud Polling
ha_codeowners:
  - '@erwindouna'
ha_domain: portainer
ha_config_flow: true
ha_platforms:
  - binary_sensor
ha_integration_type: integration
---

The Portainer {% term integration %} is used as an interface to the [Portainer API](https://docs.portainer.io/api).
Portainer is a lightweight management UI that allows you to easily manage your Docker containers, images, networks, and volumes. It works on every Docker host or Swarm cluster.

The Portainer API provides a way to manage Docker containers, images, networks, and volumes. It allows you to interact programmatically with your Docker host or Swarm cluster.

> 

Reviewer: ```suggestion

## Connect with Portainer

Before you can configure Portainer within Home Assistant, you need an Access Token. Create a Portainer Access Token by following these steps:

1. Log in to your Portainer instance.
2. Go to your user profile by clicking on your username in the top right corner and select `My Account`.
3. Goto the section `Access Tokens`.
4. Click on `Add Access Token` to create a new key.
5. Copy the generated Access Token.
6. In Home Assistant, go to the Portainer integration configuration and paste the Access Token into the appropriate field.

{% include integrations/config_flow.md %}

```

> ## Connect with Portainer

Reviewer: ```suggestion
## Prerequisites
```

Before you can configure Portainer within Home Assistant, you need an Access Token. Create a Portainer Access Token by following these steps:

1. Log in to your Portainer instance.

> 2. Go to your user profile by clicking on your username in the top right corner and select `My Account`.
> 3. Goto the section `Access Tokens`.
> 4. Click on `Add Access Token` to create a new key.

Reviewer: ```suggestion
2. To create an access token, follow the steps in the [Portainer documentation](https://docs.portainer.io/api/access).
```

> 5. Copy the generated Access Token.

Reviewer: ```suggestion
3. Copy the generated Access Token and store it somewhere safe, you will need it in the next steps.
```

> 6. In Home Assistant, go to the Portainer integration configuration and paste the Access Token into the appropriate field.

Reviewer: ```suggestion
```

{% include integrations/config_flow.md %}

## Supported functionality

{% note %}
This integration is currently being actively developed and tested. More platforms and features will be added in.
{% endnote %}

There is currently support for the following device types within Home Assistant:

- Binary sensor - for monitoring the status of Portainer services.

> 

Reviewer: ```suggestion

## Removing the integration

This integration follows standard integration removal.

{% include integrations/remove_device_service.md %}
```

> ## Connect with Portainer
> 
> Before you can configure Portainer within Home Assistant, you need an Access Token. Create a Portainer Access Token by following these steps:
> 
> 1. Log in to your Portainer instance.
> 2. Go to your user profile by clicking on your username in the top right corner and select `My Account`.
> 3. Goto the section `Access Tokens`.
> 4. Click on `Add Access Token` to create a new key.
> 5. Copy the generated Access Token.
> 6. In Home Assistant, go to the Portainer integration configuration and paste the Access Token into the appropriate field.
> 
> {% include integrations/config_flow.md %}

Reviewer: ```suggestion
```

---
title: Portainer
description: Instructions on how to integrate Portainer with Home Assistant.
ha_category:
  - Binary sensor

> ha_release: 0.41

Reviewer: ```suggestion
ha_release: 2025.7
```

ha_iot_class: Cloud Polling
ha_codeowners:
  - '@erwindouna'
ha_domain: portainer
ha_config_flow: true
ha_platforms:
  - binary_sensor
ha_integration_type: integration
---

The Portainer {% term integration %} is used as an interface to the [Portainer API](https://docs.portainer.io/api).
Portainer is a lightweight management UI that allows you to easily manage your Docker containers, images, networks, and volumes. It works on every Docker host or Swarm cluster.

The Portainer API provides a way to manage Docker containers, images, networks, and volumes. It allows you to interact programmatically with your Docker host or Swarm cluster.

## Prerequisites

> Before you can configure Portainer within Home Assistant, you need an Access Token. Create a Portainer Access Token by following these steps:

Reviewer: Are there any other components needed before I can add the integration?
Like a Portainer account?

Author: Basically, like Proxmox, you need to have it installed and up and running. Portainer does require you create an admin account via its [initial setup](https://docs.portainer.io/start/install-ce/server/setup).

> Before you can configure Portainer within Home Assistant, you need an Access Token. Create a Portainer Access Token by following these steps:

Reviewer: ```suggestion
Before you can configure Portainer within Home Assistant, you need a few things:

- have Portainer installed and a user with administrator rights on Portainer.
- an Access Token. 

Create a Portainer Access Token by following these steps:
```
something like this?

> 1. Log in to your Portainer instance.

Reviewer: We don't usually document "external" steps in the HA docs. This is why I removed a few steps and pointed to the Portainer docs instead. Could you check if these steps are still correct?

Author: This seems sufficient then. :)

2. To create an access token, follow the steps in the [Portainer documentation](https://docs.portainer.io/api/access).
3. Copy the generated Access Token and store it somewhere safe, you will need it in the next steps.

{% include integrations/config_flow.md %}

## Supported functionality

{% note %}
This integration is currently being actively developed and tested. More platforms and features will be added in.
{% endnote %}

There is currently support for the following device types within Home Assistant:

- Binary sensor - for monitoring the status of Portainer services.

## Removing the integration

> 

Reviewer: is there anything else I should be doing after removing the integration? like deleting the access token? is it specific to Home Assistant?

Author: That would be up to the users' own policy. Then can in theory use the access token in multiple external apps (not recommended). That kind of policies would be up for the user to dictate and not for HA to enforce, I believe.

Reviewer: right. let me rephrase. would it make sense to mention that maybe you want to delete the access token? as one might forget about it?

Author: That might make sense to give them the kind reminder and bright spark. ;-)

The Portainer API provides a way to manage Docker containers, images, networks, and volumes. It allows you to interact programmatically with your Docker host or Swarm cluster.

## Prerequisites

Before you can configure Portainer within Home Assistant, you need a few things:

- have Portainer installed and a user with administrator rights on Portainer.
- an Access Token. 

Create a Portainer Access Token by following these steps:

1. Log in to your Portainer instance.
2. To create an access token, follow the steps in the [Portainer documentation](https://docs.portainer.io/api/access).
3. Copy the generated Access Token and store it somewhere safe, you will need it in the next steps.

{% include integrations/config_flow.md %}

## Supported functionality

{% note %}
This integration is currently being actively developed and tested. More platforms and features will be added in.
{% endnote %}

There is currently support for the following device types within Home Assistant:

- Binary sensor - for monitoring the status of Portainer services.

## Removing the integration

This integration follows standard integration removal.

> {% include integrations/remove_device_service.md %}

Reviewer: ```suggestion
{% include integrations/remove_device_service.md %}

After removing the integration, consider deleting the Portainer access token.
```
or something like that?

---
title: Portainer
description: Instructions on how to integrate Portainer with Home Assistant.
ha_category:
  - Binary sensor

> ha_release: 2025.7

Reviewer 2: ```suggestion
ha_release: "2025.10"
```

ha_iot_class: Cloud Polling
ha_codeowners:
  - '@erwindouna'
ha_domain: portainer
ha_config_flow: true
ha_platforms:
  - binary_sensor
ha_integration_type: integration
---

The Portainer {% term integration %} is used as an interface to the [Portainer API](https://docs.portainer.io/api).
Portainer is a lightweight management UI that allows you to easily manage your Docker containers, images, networks, and volumes. It works on every Docker host or Swarm cluster.

The Portainer API provides a way to manage Docker containers, images, networks, and volumes. It allows you to interact programmatically with your Docker host or Swarm cluster.

## Prerequisites

Before you can configure Portainer within Home Assistant, you need a few things:

- have Portainer installed and a user with administrator rights on Portainer.

> - an Access Token. 

Reviewer 2: ```suggestion
- An access token. 
```
